File: pptx-service/renderers/argument.py
from builder import add_bg, add_textbox, add_rect, add_eyebrow, add_line
from theme import ThemeColors


def render(slide, data: dict, t: ThemeColors) -> None:
    add_bg(slide, t.bg)
    heading = data.get("heading", "")
    points = data.get("points", [])
    eyebrow = data.get("eyebrow", "")

    if eyebrow:
        add_eyebrow(slide, eyebrow, t)

    add_textbox(slide, 0.83, 0.8, 11.0, 1.5, heading,
                t.font_display, 52, t.text, bold=True, wrap=True)

    add_line(slide, 0.83, 2.5, 12.5, 2.5, t.border, 0.5)

    y = 2.7
    for point in points[:6]:
        add_rect(slide, 0.83, y + 0.15, 0.1, 0.1, fill_hex=t.accent)
        add_textbox(slide, 1.1, y, 11.0, 0.65, point,
                    t.font_body, 28, t.text, wrap=True)
        y += 0.75

File: lib/pptx/slides/argument.ts
import type { ArgumentSlide } from '@/lib/types';
import {
  px, py, SLIDE_W,
  setBackground, addRect, addTextBox, addHeading, addPageNumber,
  type PptxTheme,
} from '../helpers';
import type PptxGenJS from 'pptxgenjs';
type PSlide = ReturnType<InstanceType<typeof PptxGenJS>['addSlide']>;

export function render(slide: PSlide, data: ArgumentSlide, t: PptxTheme, n: number, total: number): void {
  setBackground(slide, '#' + t.bg);
  addHeading(slide, data.heading, t, py(80));

  data.points.forEach((point, i) => {
    const y = py(220) + i * py(120);
    addRect(slide, t.paddingIn, y + py(10), px(16), px(16), '#' + t.accent);
    addTextBox(slide, point, {
      x: t.paddingIn + px(32), y, w: SLIDE_W - 2 * t.paddingIn - px(32),
      fontSize: t.bodySize, color: '#' + t.text, fontFace: t.fontBody, wrap: true,
    });
  });

  addPageNumber(slide, n, total, t);
}
